### PR TITLE
fix(groups): group user list

### LINKED ISSUE #764: bug(Groups): Cannot add friend to already existing groups
**Issue:**
Like the title says you cannot add a friend to an already existing group. When you select their name it will just force unselect it 

**Screen recording:**

https://github.com/user-attachments/assets/9f081e0c-c0c1-41c0-8d4e-da701d6a8f18

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- Fix group participants inconsistency:
  - It was possible to "remove" (only visually) users by clicking on the user list instead of the buttons on top  
  
 - Resolve #764 764

### CODE REVIEW

**Comment by phillsatellite** [comment]
Hey Kevin! I think this bug fits the scope of this pr but we can handle it in another ticket if need be. So while testing it seems removing a user on the top bar will work fine but if you scroll through the list and remove them from there it doesn't actually remove them it will just reselect them 

https://github.com/user-attachments/assets/bb30b325-9e8c-424a-9c18-613d90f3890a

**Comment by Flemmli97** [comment]
> Hey Kevin! I think this bug fits the scope of this pr but we can handle it in another ticket if need be. So while testing it seems removing a user on the top bar will work fine but if you scroll through the list and remove them from there it doesn't actually remove them it will just reselect them
> 
>  Screen.Recording.2024-11-01.at.12.51.17.PM.mov

yes thats intended. the list shows ALL friends of yours. if they are selected they are in the group

**Comment by phillsatellite** [comment]
> > Hey Kevin! I think this bug fits the scope of this pr but we can handle it in another ticket if need be. So while testing it seems removing a user on the top bar will work fine but if you scroll through the list and remove them from there it doesn't actually remove them it will just reselect them
> > Screen.Recording.2024-11-01.at.12.51.17.PM.mov
> 
> yes thats intended. the list shows ALL friends of yours. if they are selected they are in the group

Yeah I see the list shows all your friends but check out the video I added again. If you try to unselect someone from the list it just re adds them quickly  same with trying to select someone from the list when you click them it'll select them for a moment and then unselect them

File: src/lib/components/group/ViewMembers.svelte [hunk 1]
```diff
@@ -21,6 +21,16 @@
     let friends: User[] = [] // Initialize friends as an empty array
 
     function update_members(user: User) {
+        if (user.key === activeChat.creator) {
+            Store.addToastNotification(new ToastMessage("", `You can not remove the group creator`, 2))
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: Add this message on our en.json file

File: src/lib/components/group/ViewMembers.svelte [hunk 2]
```diff
@@ -21,6 +21,16 @@
...
+            members = members
+            return
+        }
+        if (members.length < 3) {
+            Store.addToastNotification(new ToastMessage("", `A group can not exist with one person`, 2))
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: Add this message on our en.json file

File: src/lib/components/group/ViewMembers.svelte [hunk 3]
```diff
@@ -34,14 +44,14 @@
...
-        if (members.length < 3) {
-            Store.addToastNotification(new ToastMessage("", `A group can not exist with one person`, 2))
-            return
-        }
         if (user.key === activeChat.creator) {
             Store.addToastNotification(new ToastMessage("", `You can not remove the group creator`, 2))
             return
         }
+        if (members.length < 3) {
+            Store.addToastNotification(new ToastMessage("", `A group can not exist with one person`, 2))
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: Add this message on our en.json file

File: src/lib/components/group/ViewMembers.svelte [hunk 4]
```diff
@@ -34,14 +44,14 @@
     }
 
     function remove_member(user: User) {
-        if (members.length < 3) {
-            Store.addToastNotification(new ToastMessage("", `A group can not exist with one person`, 2))
-            return
-        }
         if (user.key === activeChat.creator) {
             Store.addToastNotification(new ToastMessage("", `You can not remove the group creator`, 2))
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: Add this message on our en.json file

**Review by lgmarchi — CHANGES_REQUESTED** [review]

**Comment by lgmarchi** [comment]
@Flemmli97 It just work to add a user to the group, but it does not work to remove user from a group. 

See the video video below to understand: 


https://github.com/user-attachments/assets/82c7f14d-3a7b-49a3-90c8-b1f4c92edc57

File: src/lib/components/group/ViewMembers.svelte [hunk 5]
```diff
@@ -21,25 +20,36 @@
...
         } else {
             group_members.push(user)
+            RaygunStoreInstance.addGroupParticipants(activeChat.id, [user.key])
         }
 
         members = group_members
-        RaygunStoreInstance.addGroupParticipants(activeChat.id, [user.key])
     }
 
     function remove_member(user: User) {
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: We use cammelCase for the name of the functions and variables.

So the correct is removeMembers.

File: src/lib/components/group/ViewMembers.svelte [hunk 6]
```diff
@@ -21,25 +20,36 @@
     let friends: User[] = [] // Initialize friends as an empty array
 
     function update_members(user: User) {
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: We use cammelCase for the name of the functions and variables.

So the correct is updateMembers.

**Comment by Flemmli97** [comment]
> @Flemmli97 It just work to add a user to the group, but it does not work to remove user from a group.
> 
> See the video video below to understand:
> 
>  Screen.Recording.2024-11-05.at.13.26.26.mov

hmm its working on my end

https://github.com/user-attachments/assets/83657f91-6dd4-438f-85af-120bcfea2643

can you post the logs?

File: src/lib/components/group/ViewMembers.svelte [hunk 7]
```diff
@@ -21,25 +20,36 @@
...
+        if (user.key === activeChat.creator) {
+            Store.addToastNotification(new ToastMessage("", $_("chat.group.removeCreator"), 2))
+            members = members
+            return
+        }
         let group_members = [...members]
```
Comment (lgmarchi) [inline src/lib/components/group/ViewMembers.svelte]: groupMembers.

File: src/lib/components/group/ViewMembers.svelte (hunk 5 above)
Comment (lgmarchi) [reply src/lib/components/group/ViewMembers.svelte]: With new code in update_members, we can remove this function and use just one.

**Review by lgmarchi — CHANGES_REQUESTED** [review]
There are some changes necessary yet.

**Comment by lgmarchi** [comment]
> > @Flemmli97 It just work to add a user to the group, but it does not work to remove user from a group.
> > See the video video below to understand:
> > Screen.Recording.2024-11-05.at.13.26.26.mov
> 
> hmm its working on my end
> 
>  2024-11-05.17-34-02.mp4 
> can you post the logs?

Never mind, probably was a cache problem here. Now it is working good.

**Review by lgmarchi — APPROVED** [review]
Code looks good and it is working as expected as well.


https://github.com/user-attachments/assets/dd8ed1fa-02ba-4a55-849e-4a79557aeab4

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://github.com/user-attachments/assets/74498dba-2234-4b2c-9b11-a063a14b6706

does not work if you are not the group creator

flow
- someone else creates the group
- try to do the above flow
- it removes and after some seconds adds right away

only works if you are the group creator

**Comment by Flemmli97** [comment]
> Gravacao.do.ecra.2024-11-05.as.23.57.38.mov 
> does not work if you are not the group creator
> 
> flow
> 
> * someone else creates the group
> * try to do the above flow
> * it removes and after some seconds adds right away
> 
> only works if you are the group creator

yes. permission got redone with new wasm update

**Comment by Flemmli97** [comment]
i would suggest we keep this pr as is as ill address the permission thing in another pr

**Comment by stavares843** [comment]
added blocked label, being discussed with flem

**Comment by stavares843** [comment]
after discussing with flem, merging as is and will fix the non admin stuff in another PR

**Review by stavares843 — APPROVED** [review]

**Pull request merged**